openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateElectricEquipmentITEAirCooled.cpp

```
@@ -0,0 +1,265 @@
+          }
+        }
+      }
+    }
+
+    // apply contraint #1: For multiple ITE objects defined for one zone, the same calculation method should apply.
+    ElectricEquipmentITEAirCooledVector electricEquipmentITEAirCooled = space->electricEquipmentITEAirCooled();
+    std::string thisMethod = definition.airFlowCalculationMethod();
+    std::sort(electricEquipmentITEAirCooled.begin(), electricEquipmentITEAirCooled.end(), WorkspaceObjectNameLess());
+    for (ElectricEquipmentITEAirCooled& iTequipment : electricEquipmentITEAirCooled) {
+      if (istringEqual(thisMethod, iTequipment.electricEquipmentITEAirCooledDefinition().airFlowCalculationMethod())) {
+        // do nothing
+      } else {
+        // if different calculation methods are assigned, all change to FlowControlWithApproachTemperatures
+        if (!istringEqual(thisMethod, "FlowControlWithApproachTemperatures")) {
+          definition.setAirFlowCalculationMethod("FlowControlWithApproachTemperatures");
+          LOG(Warn, space->briefDescription() << " has multiple IT equipment with different air flow calculation methods, " << modelObject.briefDescription() << " is re-assigned to 'FlowControlWithApproachTemperatures'");
+        }
+      }
+    }
+
+    // Contraint #2:The FlowControlWithApproachTemperatures only applies to ITE zones with single duct VAV terminal unit.
+    if (istringEqual(thisMethod, "FlowControlWithApproachTemperatures")) {
+      if (thermalZone) {
+        auto terminal = thermalZone->airLoopHVACTerminal();
+        if (terminal) {
+          if (!terminal->optionalCast<AirTerminalSingleDuctVAVReheat>() && !terminal->optionalCast<AirTerminalSingleDuctVAVNoReheat>()) {
+            LOG(Warn, modelObject.briefDescription() << " The FlowControlWithApproachTemperatures only applies to ITE zones with single duct VAV terminal unit.");
+          }
+        }
+      }
+    }
+
+  } else {
+    LOG(Warn, modelObject.briefDescription() << " not assigned to a valid space.");
```

Again is it best to not translate the object or let the simulation fail?  People may ignore a warning if the simulation does not error out......  @kbenne you have thoughts on this?

---

It seems the lights translator let the simulation fail when neither a space nor a spacetype is associated.
[https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateLights.cpp#L65](url)